pkg/cloudprovider/vsphere/vsphere_test.go

```diff
@@ -216,6 +217,113 @@ func TestVSphereLoginByToken(t *testing.T) {
 	vcInstance.Conn.Logout(ctx)
 }
 
+func TestAlphaDualStackConfig(t *testing.T) {
+	var vs *VSphere
+	var (
+		username = "user"
+		password = "password"
+	)
+	var testcases = []struct {
+		testName                 string
+		conf                     string
+		enableDualStackFeature	 bool
+		expectedIsSecretProvided bool
+		expectedUsername         string
+		expectedPassword         string
+		expectedError            error
+		expectedThumbprints      map[string]string
+	}{
+		{
+			testName: "Verifying dual stack env var required when providing two ip families",
+			conf: `[Global]
+			user = user
+			password = password
+			datacenters = us-west
+			[VirtualCenter "0.0.0.0"]
+			user = user
+			password = password
+			ip-family = ipv6,ipv4
+			
+			`,
+			enableDualStackFeature: false,
+			expectedUsername: username,
+			expectedPassword: password,
+			expectedError:    vcfg.ErrMissingDualStackEnvVar,
+		},
+		{
+			testName: "Verifying dual stack env var existing when providing two ip families",
+			conf: `[Global]
+			user = user
+			password = password
+			datacenters = us-west
+			[VirtualCenter "0.0.0.0"]
+			user = user
+			password = password
+			ip-family = ipv6,ipv4
+			
+			`,
+			enableDualStackFeature: true,
+			expectedUsername: username,
+			expectedPassword: password,
+			expectedError:    nil,
+		},
+		{
+			testName: "Dual stack env var not required when providing single ip family",
+			conf: `[Global]
+			user = user
+			password = password
+			datacenters = us-west
+			[VirtualCenter "0.0.0.0"]
+			user = user
+			password = password
+			ip-family = ipv6
+			
+			`,
+			enableDualStackFeature: false,
+			expectedUsername: username,
+			expectedPassword: password,
+			expectedError:    nil,
+		},
+	}
+	for _, testcase := range testcases {
+		t.Logf("Executing Testcase: %s", testcase.testName)
+		cfg, err := ccfg.ReadCPIConfig([]byte(testcase.conf))
+		if err != nil {
+			if testcase.expectedError != nil {
+				if err != testcase.expectedError {
+					t.Fatalf("readConfig: expected err: %s, received err: %s", testcase.expectedError, err)
+				} else {
+					continue
+				}
+			} else {
+				t.Fatalf("readConfig: unexpected error returned: %v", err)
+			}
+		}
+
+		if testcase.enableDualStackFeature {
+			_ :os.Setenv("ENABLE_ALPHA_DUAL_STACK", "1")
+		}
+
+		vs, err = buildVSphereFromConfig(cfg, nil)
+		print(err)
```

Comment:
Use `t.Log` to print the error and we should probably only print it when not nil